labelbox/data/mixins.py

```diff
@@ -1,32 +1,70 @@
-from typing import Optional
+from typing import Optional, List
 
 from pydantic import BaseModel, validator
 
-from labelbox.exceptions import ConfidenceNotSupportedException
+from labelbox.exceptions import ConfidenceNotSupportedException, CustomMetricsNotSupportedException
 
 
 class ConfidenceMixin(BaseModel):
     confidence: Optional[float] = None
 
-    @validator('confidence')
+    @validator("confidence")
     def confidence_valid_float(cls, value):
         if value is None:
             return value
         if not isinstance(value, (int, float)) or not 0 <= value <= 1:
-            raise ValueError('must be a number within [0,1] range')
+            raise ValueError("must be a number within [0,1] range")
         return value
 
     def dict(self, *args, **kwargs):
         res = super().dict(*args, **kwargs)
-        if 'confidence' in res and res['confidence'] is None:
-            res.pop('confidence')
+        if "confidence" in res and res["confidence"] is None:
+            res.pop("confidence")
         return res
 
 
 class ConfidenceNotSupportedMixin:
-
     def __new__(cls, *args, **kwargs):
-        if 'confidence' in kwargs:
+        if "confidence" in kwargs:
             raise ConfidenceNotSupportedException(
-                'Confidence is not supported for this annotaiton type yet')
+                "Confidence is not supported for this annotation type yet"
+            )
+        return super().__new__(cls)
+
+
+class CustomMetric(BaseModel):
+    name: str
+    value: float
+
+    @validator("name")
+    def confidence_valid_float(cls, value):
+        if not isinstance(value, str):
+            raise ValueError("Name must be a string")
+        return value
+
+    @validator("value")
+    def value_valid_float(cls, value):
+        if not isinstance(value, (int, float)):
+            raise ValueError("Value must be a number")
+        return value
+
+
+class CustomMetricsMixin(BaseModel):
+    custom_metrics: Optional[List[CustomMetric]] = None
+
+    def dict(self, *args, **kwargs):
+        res = super().dict(*args, **kwargs)
+
+        if "customMetrics" in res and res["customMetrics"] is None:
```

Comment:
so, to be clear, we do not want to serlaize custom metrics?

Reply:
Could you elaborate a bit please? This particular line is just making sure we don't pass None, but rather remove the key entirely in that case

Reply:
Sorry IS NONE... missed it